src/Auth0.AuthenticationApi/Tokens/IdTokenClaimValidator.cs

```diff
@@ -0,0 +1,112 @@
+﻿using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+
+namespace Auth0.AuthenticationApi.Tokens
+{
+    /// <summary>
+    /// Perform validation of a JWT ID token in compliance with specified <see cref="IdTokenRequirements"/>.
+    /// </summary>
+    internal static class IdTokenClaimValidator
```

Comment:
This is, by far, the most clear implementation of the ID token validation stuff I've seen 💪